controller/cartController.js

```diff
@@ -0,0 +1,113 @@
+import { BaseController } from "./baseController";
+const stripe = require('stripe')(process.env.STRIPE_SECRET_KEY);
+
+Object.filter = (obj, predicate) => 
+    Object.keys(obj)
+          .filter( key => predicate(obj[key]) )
+          .reduce( (res, key) => (res[key] = obj[key], res), {} );
+
+export class CartController extends BaseController {
+    constructor() {
+        this.bag = localStorage.getItem('bag') ? localStorage.getItem('bag') : {};
+        this.total = localStorage.getItem('price') ? localStorage.getItem('price') : 0;
+    }
+
+    updateLocalStorage() {
```

Comment:
local storage tren backend thi la share giua cac thread day. ben js khong support parallel nen moi thg hit any endpoint la no tao 1 thread moi -> local storage share. Change to session if possible. khong thi ksao minh cung demo 1 lan thoi =))))

Reply:
the doi sang `sessionStorage ` nhe?